Engine/AlgorithmManager.cs

```diff
@@ -884,13 +927,31 @@ private static void HandleDelistedSymbols(IAlgorithm algorithm, Delistings newDe
                 // submit an order to liquidate on market close
                 if (delisting.Type == DelistingType.Warning)
                 {
-                    delistings.Add(delisting);
+	                SubmitOrderRequest request;
+
+	                delistings.Add(delisting);
                     Log.Trace("AlgorithmManager.Run(): Security delisting warning: " + delisting.Symbol.ID);
                     var security = algorithm.Securities[delisting.Symbol];
                     if (security.Holdings.Quantity == 0) continue;
-                    var submitOrderRequest = new SubmitOrderRequest(OrderType.MarketOnClose, security.Type, security.Symbol,
-                        -security.Holdings.Quantity, 0, 0, algorithm.UtcTime, "Liquidate from delisting");
-                    var ticket = algorithm.Transactions.ProcessRequest(submitOrderRequest);
+
+                    if (security.Type == SecurityType.Option)
```

Comment:
Yes, true. I like such code snippets. Their concentration in the code shows the spots that need refactoring (adding new extension points). Like this piece signals that we need to refactor it later. This piece needs refactoring for a different reason: sending MOC orders on the last day can be error prone - many things can happen from the moment it is submitted till the end of day.